### DISCUSSION TITLE
Any way to terminal if the job does not finish before next iteration?

### DISCUSSION THREAD

**Discussion by krishnaTORQUE** [BlackDex/job_scheduler#3] [Q&A]
Any way to terminal if the job does not finish before next iteration?

```rust
// This job will run in every 1 minute
scheduler.add(Job::new("0 * * * * *".parse().unwrap(), || {
    // terminate if this job does not finish within 1 minute or next iteration.
    // rest of the code for job
}));
```

**Comment by BlackDex** [comment]
Hey, Sorry for my late reply, i didn't noticed this message.

I think that is difficult to do if the job is a long running process without any looping events to stop/kill it.
You would need to start a thread, know the thread_id/handle, and later stop/kill that thread forcefully. This works different on all the different platforms, and probably also involves unsafe code if it is possible at all. I tried some quick tests using `nix` and `pthread`, but it seems to kill the parent thread also if i send a `sigkill` to the child.

The only way i know of is using `mpsc` and run your command in a loop and break if there is a stop signal sent.

If you find a good way which is thread safe, please let me know.

**Comment by krishnaTORQUE** [comment]
Thank for the clue. That make sense.
